File: MusicManagerAPI/src/main/java/com/ldz/music/manager/model/MusicType.java
package com.ldz.music.manager.model;

import javax.persistence.*;

/**
 * Created by ldalzotto on 25/04/2017.
 */
@Entity
@Table(name = "music_type")
public class MusicType {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "id")
    private Long id;

    @ManyToOne
    private UserMusicStatus userMusicStatus;

    @Column(name = "type")
    private String type;

    @Column(name = "source_url")
    private String sourceUrl;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getSourceUrl() {
        return sourceUrl;
    }

    public void setSourceUrl(String sourceUrl) {
        this.sourceUrl = sourceUrl;
    }

    public UserMusicStatus getUserMusicStatus() {
        return userMusicStatus;
    }

    public void setUserMusicStatus(UserMusicStatus userMusicStatus) {
        this.userMusicStatus = userMusicStatus;
    }
}

File: MusicManagerAPI/src/scala/com/ldz/music/manager/model/MusicType.scala
package com.ldz.music.manager.model

import javax.persistence._

import com.ldz.enumeration.ExternalMusicKey

/**
  * Created by Loic on 20/06/2017.
  */
@Entity
@Table(name = "music_type")
class MusicType {

  @Id
  @GeneratedValue(strategy = GenerationType.AUTO)
  @Column(name = "id") private var id = 0L

  @ManyToOne private var userMusicStatus: UserMusicStatus = _

  @Column(name = "type") private var `type`: String = _

  @Column(name = "source_url") private var sourceUrl : String = _

  @ElementCollection(targetClass = classOf[ExternalMusicKey.Value])
  @MapKeyColumn(name = "musicParameters")
  @Column(name = "musicParamKey")
  private var musicParameters: java.util.Map[ExternalMusicKey.Value, String] = _

  def getId: Long = id

  def setId(id: Long): Unit = {
    this.id = id
  }

  def getType: String = `type`

  def setType(`type`: String): Unit = {
    this.`type` = `type`
  }

  def getSourceUrl: String = sourceUrl

  def setSourceUrl(sourceUrl: String): Unit = {
    this.sourceUrl = sourceUrl
  }

  def getUserMusicStatus: UserMusicStatus = userMusicStatus

  def setUserMusicStatus(userMusicStatus: UserMusicStatus): Unit = {
    this.userMusicStatus = userMusicStatus
  }

  def getMusicParameters: java.util.Map[ExternalMusicKey.Value, String] = musicParameters

  def setMusicParameters(musicparameters : java.util.Map[ExternalMusicKey.Value, String]) = {
    this.musicParameters = musicparameters
  }


}
